### ISSUE TITLE
Is the listen config parameter supposed to listen to localhost?

### ISSUE THREAD

**Issue by tal-zvon** [acme-dns/acme-dns#246]
The sample config file says: 

    listen = "127.0.0.1:53"

Should that not be `0.0.0.0:53`? Why does it only need to listen to requests on the local host? Am I missing something?
Or is that just an example, and you're meant to replace `127.0.0.1` with `0.0.0.0`?

**Comment by HausnerR** [comment]
IMHO it's a bug in README and example config.

I successfully configured acme-dns making DNS server listening on `0.0.0.0:53` and making API listen only in local subnet.

So only I have access to API to register new endpoints and Let's encrypt can reach DNS server to confirm domain owner.

**Comment by tal-zvon** [comment]
> IMHO it's a bug in README and example config.
> 
> I successfully configured acme-dns making DNS server listening on `0.0.0.0:53` and making API listen only in local subnet.
> 
> So only I have access to API to register new endpoints and Let's encrypt can reach DNS server to confirm domain owner.

That makes sense. The README and example config do not. Sounds like a documentation bug to me (or whatever that would be called).

**Comment by joohoi** [comment]
It’s actually intentional. On many configurations systemd resolved is
listening at 127.0.1.53:53 which makes acme-dns unable to bind to 0.0.0.0:53

**Comment by tal-zvon** [comment]
Would not binding to 127.0.0.1:53 (as the documentation suggests) be the equivalent to binding to any other random IP? Just like binding to 1.2.3.4:53, it would technically allow the program to start (since the binding would succeed), but wouldn't allow LetsEncrypt to reach it. As far as I can tell, there's no point in listening to 127.0.0.1:53 since LetsEncrypt requests will never come from localhost, and will never hit that. Am I wrong?

If that's true, you might as well say `listen = "w.x.y.z:53"`, and just have a note to tell people to type in the IP of the interface they want to listen on. At least users would know that's something they need to figure out, rather than type in 127.0.0.1:53 as is and hope it works.

**Comment by joohoi** [comment]
Yeah, you are right. It would probably serve the users better as far as the documentation goes. Do you want to create a PR to fix it or should I do it?

**Comment by tal-zvon** [comment]
If you've got the time, go ahead.

Thanks

**Comment by alonebfg** [comment]
Hi all I hit this issue today and it was a nightmare I have tried everything the only workaround I have found to get it working is to run systemctl stop systemd-resolved.service and then disable it. and then change the interface to to 0.0.0.0:53 or :53 both of which then seem to work but I cant seem to find any other way around it.
